mailbox/event/event-cassandra/src/test/java/org/apache/james/mailbox/events/CassandraEventDeadLettersHealthCheckTest.java

```diff
@@ -0,0 +1,70 @@
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.events;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.james.backends.cassandra.CassandraCluster;
+import org.apache.james.backends.cassandra.CassandraClusterExtension;
+import org.apache.james.backends.cassandra.DockerCassandra;
+import org.apache.james.core.healthcheck.Result;
+import org.apache.james.event.json.EventSerializer;
+import org.apache.james.mailbox.model.TestId;
+import org.apache.james.mailbox.model.TestMessageId;
+import org.apache.james.mailbox.store.quota.DefaultUserQuotaRootResolver;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+
+class CassandraEventDeadLettersHealthCheckTest implements EventDeadLettersHealthCheckContract {
+    @RegisterExtension
+    static CassandraClusterExtension cassandraClusterExtension = new CassandraClusterExtension(CassandraEventDeadLettersModule.MODULE);
+    private EventDeadLettersHealthCheck testee;
+    private CassandraEventDeadLetters eventDeadLetters;
+
+    @BeforeEach
+    void setUp(CassandraCluster cassandraCluster) {
+        EventSerializer eventSerializer = new EventSerializer(new TestId.Factory(), new TestMessageId.Factory(), new DefaultUserQuotaRootResolver.DefaultQuotaRootDeserializer());
+        eventDeadLetters = new CassandraEventDeadLetters(new CassandraEventDeadLettersDAO(cassandraCluster.getConf(), eventSerializer),
+                                                         new CassandraEventDeadLettersGroupDAO(cassandraCluster.getConf()));
+        testee = new EventDeadLettersHealthCheck(eventDeadLetters);
+    }
+
+    @Override
+    public EventDeadLettersHealthCheck testee() {
+        return testee;
+    }
+
+    @Override
+    public EventDeadLetters eventDeadLetters() {
+        return eventDeadLetters;
+    }
+
+    @Test
+    public void checkShouldReturnUnHealthyWhenEventDeadLetterErrorWithCassandraConnection(DockerCassandra dockerCassandra) {
```

Comment:
why it's not in the contract testsuite?

Reply:
be cause this case is can be happened with `Cassandra version` but not `memory version`.So i dont want `memory version` had unnecessary test case.

Reply:
untrue, the code exists for both backends, you are just unable to trigger the catch clause with memory.
However, it should be in the contract and disabled for Memory because memory is more the exception than the rule.

Reply:
I got it.